Issue 7126 - "saferepr" to avoid BytesWarning when using --setup-show (#7205)
pytest-dev/pytest: changelog/7126.bugfix.rst

> ``--setup-show`` now doesn't raise an error if bytearray is used as ``parametrize``
> parameter when Python is called with ``-bb`` flag.

Reviewer: ```suggestion
``--setup-show`` now doesn't raise an error when a bytes value is used as a ``parametrize``
parameter when Python is called with the ``-bb`` flag.
```

> Switched to ``saferepr`` to format bytes ``parametrize`` parameters

Reviewer: `saferepr` is an internal name, so people reading this are unlikely to know what it means. I suggest rephrasing to only describe the bug that is fixed, leaving out the actual solution.

for ``--setup-show`` output to prevent errors when Python is called with ``-bb``

> to catch bytearray with unicode comparison.

Reviewer: This line is not accurate in this case -- `-bb` errors not due to comparison but because of `str(b'abc')`.

Overall I don't think the cause is very important, only the fact that it failed. So I'd omit the "to catch ..." part.
